collect_app/src/main/java/org/odk/collect/android/tasks/sms/models/SentMessageResult.java

```diff
@@ -0,0 +1,73 @@
+package org.odk.collect.android.tasks.sms.models;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+/**
+ * Encapsulates all the data that's received when the SmsSentBroadcastReceiver
+ * is triggered. This is then passed to the SmsService and Notification Receiver for processing.
+ */
+public class SentMessageResult implements Parcelable {
```

Comment:
Okay! I created it as a means of passing results from the sent and delivery report receivers. The idea behind that decision was I wanted to use a common object that would be able to store the combined state of the two and so the layers that neded the results wouldn't need to deal with the result codes. `SmsNotificationReceiver` and `SmsService` both use the `instanceId` and the `resultCode` which is represented by the enum.

Reply:
@lognaturel what are thoughts on how many arguments should be passed to a function? When would you create a class to store them?

Reply:
I'm also not thrilled by having this implement the `Parcelable` interface when really all you need to pass with the intent is an int. That feels like an obfuscation to me, does that make sense?

Reply:
Okay, I understand where you are coming from. Thanks!

Reply:
Also something like `smsSubmissionManager.updateMessageStatus(toMessageStatus(sentMessageResult.getMessageResultStatus()), sentMessageResult.getInstanceId(), sentMessageResult.getMessageId());` where you're breaking it up makes me think you found the signature for `updateMessageStatus` that took the individual values clearer. I do tend to agree and think it would be even better without all the `sentMessageResult.` calls.

Reply:
Yes, that makes sense. It's just habits that have caused some of these coding styles. Learning that when passing data from one class to the next it's easy to use a `Parcelable` and fast as well eg. I use a plugin that generates it and I put all the fields in the class so once I generate the getters and setters and the plugin generates the `Parcelable` and that's passed then I wouldn't have to worry about getting each value from intent one by one. So now, i am looking at things alot differently!